### PR TITLE
support arbitrary dimensions in combinedims

### PR DESCRIPTION
**Pull request by aplavin**

### CODE REVIEW

**Comment by aplavin** [comment]
Many `@inferred` tests fail on 1.0. Maybe it's fine?

**Comment by andyferris** [comment]
Sorry I haven't looked at this yet!

> Many @inferred tests fail on 1.0. Maybe it's fine?

I think now 1.7 is out it is reasonable to move forward with Julia 1.6 as the minimum supported Julia version on these sorts of packages.

File: src/combinedims.jl [hunk 1]
```diff
@@ -16,23 +16,33 @@ julia> combinedims([[1, 2], [3, 4]])
...
-            out[i.I..., j.I...] = a[j][i]
-        end
+    for j in CartesianIndices(outeraxes)
+        I = slice_inds(j, Val(outer_dims), Val(ndims_total))
+        out[I..., :] = a[j]  # trailing ':' is required for zero-dimensional a[j]
     end
     return out
 end
 
+@inline function _combine_tuples(n, dims, t_then, t_else)
```
Comment (andyferris) [inline src/combinedims.jl:38]: Out of curiousity - does this support constant propagation? Did the old code? I wonder if there is a change with respect to a type like `StaticArrays.MArray`?

**Review by andyferris — APPROVED** [review]
Awesome, looks great. It's wonderful to have someone else also looking at this code, so thank you!

File: src/combinedims.jl (hunk 1 above)
Comment (aplavin) [reply src/combinedims.jl:38]: The `combinedims` function without dimensions does infer, with dimensions - not completely... Gives a 2-union.
See https://github.com/JuliaData/SplitApplyCombine.jl/pull/49

**Pull request merged**